app/controllers/api/greetings_controller.rb

```diff
@@ -0,0 +1,6 @@
+# app/controllers/messages_controller.rb
+class MessagesController < ApplicationController
```

Comment:
Great effort on your `MessagesController`! However, it seems there might be a slight mix-up in naming conventions. Since the file is named `greeting_controller.rb` in the `controllers/api` directory, the class should follow the CamelCase convention and be named `GreetingsController`. This is causing an error of `uninitialized constant Api::GreetingsController`

The error "uninitialized constant Api::GreetingsController" indicates a mismatch between the expected controller name and the actual controller name in your Rails application.

If your file is named `greeting_controller.rb`, Rails would expect the class to be named `GreetingsController`, not `MessagesController`. Let's update your controller file to match the expected naming convention:

```ruby
# app/controllers/api/greetings_controller.rb
module Api
  class GreetingsController < ApplicationController
    def random_greeting
      render json: { greeting: Message.order('RANDOM()').first&.content }
    end
  end
end
```

Ensure that your routes are configured correctly in `config/routes.rb`:

```ruby
namespace :api do
  resources :greetings do
    get :random_greeting, on: :collection
  end
end
```

This assumes you want to handle the `random_greeting` action within the `Api::GreetingsController`. Adjust the routes according to your application's requirements.

Remember to restart your Rails server after making these changes:

```bash
rails server
```

Now, you should be able to access the `random_greeting` action at the appropriate route in your API. For example, if you have the default Rails server running on `localhost:3000`, you can access it at `http://localhost:3000/api/greetings/random_greeting`.